include/deal.II/base/types.h

```diff
@@ -150,6 +150,11 @@ namespace types
    * @ref GlossMaterialId "Glossary entry on material indicators"
    */
   using material_id = unsigned int;
+
+  /**
+   * The type used to denote geometric entity types.
+   */
+  using geometric_entity_type = std::uint8_t;
```

Comment:
In reference to the other PR: How about `reference_cell_kind` or `reference_cell_type`?

Reply:
Separately, though: Where do you actually need this? I believe that the only place where you use it is in the definition of the `enum class`. Nowhere else should we actually be using this. In that case, let's either make this an internal namespace type or, better, just omit giving it a name altogether: just put it into the declaration of the `enum class` directly.

Reply:
I am using it to cast the enum to a number in #10631 so that I can access an entry in an array. But I guess implement it there also differently!

Reply:
OK, I have remove `geometric_entity_type` and rename the enum to `RefernceCell::Type`!